beacon-chain/slasher/detect_attestations.go

```diff
@@ -120,6 +159,79 @@ func (s *Service) determineChunksToUpdateForValidators(
 	return
 }
 
+// Checks if an incoming attestation is slashable based on the validator chunk it
+// corresponds to. If a slashable offense is found, we return it to the caller.
+// If not, then update every single chunk the attestation covers, starting from its
+// source epoch up to its target.
+func (s *Service) applyAttestationForValidator(
+	ctx context.Context,
+	opts *chunkUpdateOptions,
+	chunksByChunkIdx map[uint64]Chunker,
+	attestation *slashertypes.CompactAttestation,
+) (*slashertypes.Slashing, error) {
+	sourceEpoch := attestation.Source
+	targetEpoch := attestation.Target
+	chunkIdx := s.params.chunkIndex(sourceEpoch)
+	chunk, ok := chunksByChunkIdx[chunkIdx]
+	if !ok {
+		return nil, nil
+	}
+
+	// Check slashable, if so, return the slashing.
+	slashing, err := chunk.CheckSlashable(
+		ctx,
+		s.serviceCfg.Database,
+		opts.validatorIndex,
+		attestation,
+	)
+	if err != nil {
+		return nil, err
+	}
+	if slashing != nil && slashing.Kind != slashertypes.NotSlashable {
+		return slashing, nil
+	}
+
+	// Get the first start epoch for the chunk. If it does not exist or
+	// is not possible based on the input arguments, do not continue with the update.
+	startEpoch, exists := chunk.StartEpoch(sourceEpoch, opts.currentEpoch)
+	if !exists {
+		return nil, nil
+	}
+
+	// Given a single attestation could span across multiple chunks
+	// for a validator min or max span, we attempt to update the current chunk
+	// for the source epoch of the attestation. If the update function tells
+	// us we need to proceed to the next chunk, we continue by determining
+	// the start epoch of the next chunk. We exit once no longer need to
+	// keep updating chunks.
+	for {
+		chunkIdx = s.params.chunkIndex(startEpoch)
+		chunk, ok = chunksByChunkIdx[chunkIdx]
+		if !ok {
+			return nil, nil
```

Comment:
Again, shouldn't this be an error? We already checked that the `startEpoch` exists in `if !exists`.